### PR TITLE
Added MOver tests, and split steps by supplier rather than feature

### PR DESCRIPTION
**Pull request by KieranWardle**

### CODE REVIEW

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@Slf4j
+@PropertySource("classpath:application.properties")
+public class AdeccoSteps {
+
+  public static AdeccoResponse adeccoResponse = new AdeccoResponse();
+  public static List<AdeccoResponse> adeccoResponseList = new ArrayList<>();
+
+  @Given("An employee exists in {string} with an id of {string}")
+  public void we_recieve_an_employee_with_an_id_of(String source, String id) {
```
Comment (garethTurner) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java]: source isn't used

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  @Given("a contract start date of {string}")
+  public void a_contract_start_date_of(String date) {
+    adeccoResponse.setContractStartDate(date);
+  }
+
+  @When("we receive a job role update from adecco for employee  {string}")
+  public void we_receive_a_job_role_update_from_adecco_for_employee(String id) {
+    adeccoResponse.setContractStartDate(LocalDate.now().toString());
```
Comment (garethTurner) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java]: is this what this method should be doing? doesn't seem related to the name

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    moverResponse.setStatus("ASSIGNED");
+    moverResponse.setCrStatus("ACTIVE");
+    moverResponse.setOperationalEndDate(adeccoResponse.getOperationalEndDate());
+    moverResponse.setContractStartDate(adeccoResponse.getContractStartDate());
+    moverResponse.setContractEndDate(adeccoResponse.getContractEndDate());
+    adeccoResponseList.add(moverResponse);
+  }
+
+  @Given("their old job role {string} gets cancelled")
+  public void their_old_job_role_gets_cancelled(String string) {
```
Comment (garethTurner) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java]: "string" isn't used

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/LwsSteps.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+  @Value("${lws.operatorInstructionThreeNisra}")
+  private String operatorInstructionThreeNisra;
+
+  @Then("the employee {string} in the LWS CSV as mover with {string}")
+  public void the_employee_in_the_LWS_CSV_as_mover_with(String inCsv, String roleId) throws Exception {
+    String csvFilename = sftpUtils.getLWSFileName();
+    if (inCsv.equals("is")) {
+      String csv = sftpUtils.getCsv("lws/", csvFilename);
+      if (roleId.length() == 4) {
```
Comment (garethTurner) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/LwsSteps.java]: should make these numbers constants

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java (hunk 1 above)
Comment (spdiaz1000) [reply src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/AdeccoSteps.java]: Actually, I just saw the feature file, leave the **source** there.

File: src/test/resources/acceptancetests/creates.feature [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    And we ingest them
+    When the employee is sent to all downstream services
+    Then the employee is correctly created in gsuite with roleId "CAR1" and orgUnit "Managers"
+    And the employee is correctly created in ServiceNow with "CAR1"
+    And the employee from "ADECCO" with roleId "CAR1" is correctly created in XMA with group "7DD2611D-F60D-4A17-B759-B021BC5C669A"
+    And the employee is not in the LWS CSV as a create
+    And the employee "is" in the Logisitics CSV with "CAR1" as a create
+    And Check the employee "123456789" is sent to RCA
+
+  Scenario: A record with a start date less than 6 days in the future is not created in the downstream systems
```
Comment (spdiaz1000) [inline src/test/resources/acceptancetests/creates.feature]: should read that it **IS** created

File: src/test/resources/acceptancetests/creates.feature [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    And we ingest them
+    When the employee is sent to all downstream services
+    Then the employee "123456789" is not created in gsuite
+    And the employee is not created in ServiceNow
+    And the employee  is not created in XMA
+    And the employee is not in the LWS CSV as a create
+    And the employee is not in the Logisitics CSV
+    And Check the employee "123456781" is not sent to RCA
+
+  Scenario: A record with a start date 6 days in the future is not created in the downstream systems
```
Comment (spdiaz1000) [inline src/test/resources/acceptancetests/creates.feature]: should read that it IS created

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/utils/GsuiteMockUtils.java [hunk 7]
```diff
@@ -45,6 +47,24 @@ public void clearMock() throws IOException {
         return responseEntity.getBody();
     }
 
+//    public List<String> getGroups() {
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/utils/GsuiteMockUtils.java]: remove commented out code

File: src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/GSuiteSteps.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  private String rabbitPassword;
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Then("the employee is correctly created in gsuite with roleId {string} and orgUnit {string}")
+  public void the_employee_is_correctly_updated_in_gsuite(String roleId, String orgUnit) {
+    String[] records = gsuiteMockUtils.getRecords();
+
+    assertThat(records[0]).contains(
+        "\"changePasswordAtNextLogin\":true,\"customSchemas\":{\"Employee_Information\":{\"RoleID\":\"" + roleId
```
Comment (spdiaz1000) [inline src/test/java/uk/gov/ons/fsdr/tests/acceptance/steps/GSuiteSteps.java]: comparing json as strings makes the tests brittle as order of fields and whitespace are involved in the comparison,, better way is to have the as  json objects (you could read the strings) , and then do the compare on the jsomn object.
https://collaborate2.ons.gov.uk/jira/browse/FWMT-1875
Dont fix it in this code review Ive created this ticket.
